lib/multicell_location/services/here_integration.c

```diff
@@ -0,0 +1,230 @@
+/*
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr.h>
+#include <modem/lte_lc.h>
+#include <cJSON.h>
+
+#include "location_service.h"
+
+#include <logging/log.h>
+
+LOG_MODULE_REGISTER(multicell_location_here, CONFIG_MULTICELL_LOCATION_LOG_LEVEL);
+
+#define HOSTNAME	CONFIG_MULTICELL_LOCATION_HERE_HOSTNAME
+
+#if IS_ENABLED(CONFIG_MULTICELL_LOCATION_HERE_V1)
+#define PATH		"/positioning/v1/locate"
+#elif IS_ENABLED(CONFIG_MULTICELL_LOCATION_HERE_V2)
+#define PATH		"/v2/locate"
+#endif
+
+#if IS_ENABLED(CONFIG_MULTICELL_LOCATION_HERE_USE_API_KEY)
+BUILD_ASSERT(sizeof(CONFIG_MULTICELL_LOCATION_HERE_API_KEY) > 1, "API key must be configured");
+#define AUTHENTICATION	"apiKey="CONFIG_MULTICELL_LOCATION_HERE_API_KEY
+#elif IS_ENABLED(CONFIG_MULTICELL_LOCATION_HERE_USE_APP_CODE_ID)
+BUILD_ASSERT(sizeof(API_APP_CODE) > 1, "App code must be configured");
+BUILD_ASSERT(sizeof(API_APP_ID) > 1, "App ID must be configured");
+#define AUTHENTICATION							\
+	"app_code="CONFIG_MULTICELL_LOCATION_HERE_APP_CODE		\
+	"&app_id="CONFIG_MULTICELL_LOCATION_HERE_APP_ID
+#endif
+
+#define HTTP_REQUEST_HEADER						\
+	"POST "PATH"?"AUTHENTICATION" HTTP/1.1\r\n"			\
+	"Host: "HOSTNAME"\r\n"					        \
+	"Content-Type: application/json\r\n"				\
+	"Content-Length: %d\r\n\r\n"
+
+#define HTTP_REQUEST_BODY						\
+	"{"								\
+		"\"lte\":["						\
+			"{"						\
+				"\"mcc\": %d,"				\
+				"\"mnc\": %d,"				\
+				"\"cid\": %d,"				\
+				"\"nmr\":["				\
+					"%s"				\
+				"]"					\
+			"}"						\
+		"]"							\
+	"}"
+
+#define HTTP_REQUEST_BODY_NO_NEIGHBORS					\
+	"{"								\
+		"\"lte\":["						\
+			"{"						\
+				"\"mcc\": %d,"				\
+				"\"mnc\": %d,"				\
+				"\"cid\": %d"				\
+			"}"						\
+		"]"							\
+	"}"
+
+#define HTTP_REQUEST_BODY_NEIGHBOR_ELEMENT				\
+        "{"								\
+            "\"earfcn\": %d,"						\
+            "\"pci\": %d"						\
+        "}"
+
+BUILD_ASSERT(sizeof(HOSTNAME) > 1, "Hostname must be configured");
+
+static char body[1024];
+static char neighbors[512];
+static const char tls_certificate[] =
```

Comment:
Do we have information what certificate is this? Probably worth to mention in the comment.

Reply:
Good idea, added metadata